internal/api/tests/user_login_handler_test.go

```diff
@@ -0,0 +1,141 @@
+package handler
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"net/http"
+	"net/http/httptest"
+	"reflect"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/joshsoftware/profile_builder_backend_go/internal/api/handler"
+	"github.com/joshsoftware/profile_builder_backend_go/internal/app/service/mocks"
+	"github.com/joshsoftware/profile_builder_backend_go/internal/pkg/dto"
+	"github.com/joshsoftware/profile_builder_backend_go/internal/pkg/helpers"
+	"github.com/stretchr/testify/assert"
+	"github.com/undefinedlabs/go-mpatch"
+)
+
+func TestUserLoginHandler(t *testing.T) {
+	mockUserLoginService := new(mocks.Service)
+	handlerFunc := handler.Login(context.Background(), mockUserLoginService)
+
+	tests := []struct {
+		Name               string
+		Email              string
+		AccessToken        string
+		MockSendRequest    func(context.Context, string, string) ([]byte, error)
+		MockSetup          func(*mocks.Service, string)
+		RequestBody        dto.UserLoginRequest
+		ExpectedStatusCode int
+		ExpectedResponse   string
+	}{
+		{
+			Name:        "success",
+			Email:       "test@example.com",
```

Comment:
As we discuss in morning, can we define this vars at top of the file?

Reply:
done